From 67d4fa930d479e439e06621e29441ba3cd36b93d Mon Sep 17 00:00:00 2001
From: Wojciech Bartczak <wbartczak@marvell.com>
Date: Fri, 19 Mar 2021 07:58:42 -0700
Subject: [PATCH 1358/1921] scmi: driver: modifies rx buffer size and timeout
 for rx response

Various platforms have different requirements for timings and message
size when processing the messages.

In case of Octeon TX, the time is for message processing is consiredably
higher than default set in the driver.

SCMI cpu-freq driver allows MAX_OPPS (16) OPPs. But when transferring
the 16 OPP data from SCP to ARM, because of max_msg_size limit,
it fails to transfer information about all 16 OPPs.
Increasing the max_msg_size prevents the problem to occur and ensures
correct data reception on Linux side.

Reviewed-on: https://sj1git1.cavium.com/19451 (timeout increase)
Reviewed-on: https://sj1git1.cavium.com/16842 (max_msg_size increase)
Change-Id: If2cb0ec7206bdbb9fb2695bab9c97e357a626d7f
Signed-off-by: Wojciech Bartczak <wbartczak@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48454
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/firmware/arm_scmi/driver.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/firmware/arm_scmi/driver.c b/drivers/firmware/arm_scmi/driver.c
index b49c9e6f4bf1..0f8d58af8d4a 100644
--- a/drivers/firmware/arm_scmi/driver.c
+++ b/drivers/firmware/arm_scmi/driver.c
@@ -606,9 +606,9 @@ int scmi_handle_put(const struct scmi_handle *handle)
 }
 
 static const struct scmi_desc scmi_generic_desc = {
-	.max_rx_timeout_ms = 30,	/* We may increase this if required */
+	.max_rx_timeout_ms = 300,	/* We may increase this if required */
 	.max_msg = 20,		/* Limited by MBOX_TX_QUEUE_LEN */
-	.max_msg_size = 128,
+	.max_msg_size = 256,
 };
 
 /* Each compatible listed below must have descriptor associated with it */
-- 
2.31.1

